contracts/ConditionalTaker.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { TokenCollector } from "./abstracts/TokenCollector.sol";
+import { Ownable } from "./abstracts/Ownable.sol";
+import { EIP712 } from "./abstracts/EIP712.sol";
+import { IWETH } from "./interfaces/IWETH.sol";
+import { IConditionalTaker } from "./interfaces/IConditionalTaker.sol";
+import { ILimitOrderSwap } from "./interfaces/ILimitOrderSwap.sol";
+import { LimitOrder, getLimitOrderHash } from "./libraries/LimitOrder.sol";
+import { AllowFill, getAllowFillHash } from "./libraries/AllowFill.sol";
+import { Constant } from "./libraries/Constant.sol";
+import { Asset } from "./libraries/Asset.sol";
+import { SignatureValidator } from "./libraries/SignatureValidator.sol";
+
+/// @title ConditionalTaker Contract
+/// @author imToken Labs
+contract ConditionalTaker is IConditionalTaker, Ownable, TokenCollector, EIP712 {
```

Comment:
nit: Maybe call this `CoordinatorTaker`, since conditional taker is more about the name of role in the architecture.

Reply:
Renamed in 84c3b58.